backend/tests/endpoints/test_message.py

```diff
@@ -92,74 +150,211 @@ async def test_send_message_successful(mock_get_user_room, mock_write, mock_cent
+):
     """Send message unsuccessful when writing to zc core return a response with status_code.
 
     Args:
         mock_get_room (AsyncMock): Asynchronous external api call
-        mock_write (AsyncMock): Asynchronous external api call
+        mock_dataStorage_write (AsyncMock): Asynchronous external api call
     """
-
+    db = DataStorage("619ba4")
+    db.plugin_id = "34453"
     write_response = {"status_code": 422, "message": "unprocessible error"}
-    mock_get_user_room.return_value = fake_core_data
-    mock_write.return_value = write_response
-    response = client.post(send_message_test_url, json=test_payload)
+    mock_dataStorage_read.return_value = fake_core_room_data
+    mock_dataStorage_write.return_value = write_response
+    response = client.post(send_message_test_url, json=send_message_test_payload)
     assert response.status_code == 424
     assert response.json() == {"detail": {"Message not sent": write_response}}
+
+
+@pytest.mark.asyncio
+@mock.patch.object(DataStorage, "__init__", lambda x, y: None)
+async def test_update_message_successful(
+    mock_dataStorage_read, mock_dataStorage_update, mock_centrifugo
+):
+    """Update message successful.
+
+    Args:
+        mock_dataStorage_read (AsyncMock): Asynchronous external api call
+        mock_dataStorage_update (AsyncMock): Asynchronous external api call
+        mock_centrifugo (AsyncMock): Asynchronous external api call
+    """
+    db = DataStorage("619ba4")
+    db.plugin_id = "34453"
+    mock_dataStorage_read.return_value = fake_zc_core_message_data
+    mock_dataStorage_update.return_value = {
+        "status": 200,
+        "message": "success",
+        "data": {"matched_documents": 1, "modified_documents": 1},
+    }
+    mock_centrifugo.return_value = {"status_code": 200}
+    response = client.put(update_message_test_url, json=update_message_test_payload)
+    assert response.status_code == 200
+    assert response.json() == {
+        "status": "success",
+        "message": "Message edited",
+        "data": {
+            "_id": "346556",
+            "created_at": "2021-12-28 19:27:13.620083",
+            "edited": True,
+            "emojis": [],
+            "files": [],
+            "org_id": "1234",
+            "richUiData": {
+                "blocks": [
+                    {
+                        "data": {},
+                        "depth": 0,
+                        "entityRanges": [],
+                        "inlineStyleRanges": [],
+                        "key": "eljik",
+                        "text": "HI,  mark",
+                        "type": "unstyled",
+                    }
+                ],
+                "entityMap": {},
+            },
+            "room_id": "343235",
+            "saved_by": [],
+            "sender_id": "619ba4",
+            "threads": [],
+            "timestamp": 0,
+        },
+    }
+
+
+@pytest.mark.asyncio
+@mock.patch.object(DataStorage, "__init__", lambda x, y: None)
```

Comment:
New things to learn here. 
Are we mocking only the __init__ method or the whole DataStorage class?

Reply:
This is only mocking the init method in the DataStorage class boss